src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -202,14 +205,22 @@
                 for (LibraryRecord library : action.getLibraries()) {
                     FilePath f = libs.child(library.name + "/resources/" + name);
                     if (f.exists()) {
-                        resources.put(library.name, f.readToString());
+                        resources.put(library.name, readResource(f, encoding));
                     }
                 }
             }
         }
         return resources;
     }
 
+    private static String readResource(FilePath file, @CheckForNull String encoding) throws IOException, InterruptedException {
+        if ("Base64".equals(encoding)) {
+            return Base64.encodeBase64String(IOUtils.toByteArray(file.read()));
+        } else {
+            return IOUtils.toString(file.read(), encoding);
```

Comment:
Worth mentioning that the downstream method is null-safe, which is how we handle null encodings. 

@jglick WDYT wrt the handling of on-master vs. on-build-agent encoding here?

Reply:
Worth including a javadoc explaining handling of null/empty (i.e.: null uses default encoding for this system, and empty string is coerced to null).

Reply:
Ok, I also need to mention the Base64 encoding in the docs.

Reply:
This maintains current behavior where we use the default encoding, so I think it is OK from a compatibility perspective.